src/main/java/net/minecraftforge/fml/client/GuiModsMissing.java

```diff
@@ -38,27 +41,32 @@ public GuiModsMissing(MissingModsException modsMissing)
     public void drawScreen(int mouseX, int mouseY, float partialTicks)
     {
         this.drawDefaultBackground();
-        int offset = Math.max(85 - modsMissing.missingMods.size() * 10, 10);
+        List<Pair<ArtifactVersion, ArtifactVersion>> missingModsVersions = modsMissing.getMissingModsVersions();
+        int offset = Math.max(85 - missingModsVersions.size() * 10, 10);
         String modMissingDependenciesText = I18n.format("fml.messages.mod.missing.dependencies", TextFormatting.BOLD + modsMissing.getModName() + TextFormatting.RESET);
         this.drawCenteredString(this.fontRenderer, modMissingDependenciesText, this.width / 2, offset, 0xFFFFFF);
         offset+=10;
         String fixMissingDependenciesText = I18n.format("fml.messages.mod.missing.dependencies.fix", modsMissing.getModName());
         this.drawCenteredString(this.fontRenderer, fixMissingDependenciesText, this.width / 2, offset, 0xFFFFFF);
         offset+=5;
-        for (ArtifactVersion v : modsMissing.missingMods)
+        for (Pair<ArtifactVersion, ArtifactVersion> versionInfo : missingModsVersions)
         {
-            offset+=10;
-            if (v instanceof DefaultArtifactVersion)
+            ArtifactVersion neededVersion = versionInfo.getKey();
+            String neededMod = neededVersion.getLabel();
+            ArtifactVersion haveVersion = versionInfo.getValue();
+            String missingReason = haveVersion == null ? "missing" : "you have " + haveVersion.getVersionString();
```

Comment:
Internationalization support?

Reply:
Unfortunately this has the same problem as explained in https://github.com/MinecraftForge/MinecraftForge/pull/4761#issuecomment-367566310

Reply:
Isn't this Gui purely client-side?

Reply:
Oh right, this isn't the same problem at all. Sorry I got this mixed up with the other PR.